src/commands/react-native-ci-cli.ts

```diff
@@ -22,6 +28,57 @@ export type CycliCommand = GluegunCommand & {
   featureOptions: Option[]
 }
 
+const RECIPES = [lint, jest, typescriptCheck, prettierCheck, easUpdate, detox]
+
+const getSelectedOptions = async (toolbox: CycliToolbox): Promise<string[]> => {
+  if (toolbox.options.isPreset()) {
+    const featureFlags = RECIPES.map((option) => option.meta.flag)
+
+    const selectedOptions = intersection(
+      featureFlags,
+      Object.keys(toolbox.parameters.options)
+    )
+
+    RECIPES.forEach((recipe: CycliRecipe) => {
+      if (selectedOptions.includes(recipe.meta.flag)) {
+        try {
+          recipe.validate?.(toolbox)
+        } catch (error: unknown) {
+          const validationError = messageFromError(error)
+          throw Error(
+            `Cannot generate ${recipe.meta.name} workflow in your project.\nReason: ${validationError}`
```

Comment:
I wonder if there is any benefit in catching it and throwing again (?)

Reply:
I wanted to stick to the fact that all errors terminating the script execution are catched here

https://github.com/software-mansion-labs/react-native-ci-cli/blob/2c647fe3e999f82f30245ff5fa2114ee32be7313/src/commands/react-native-ci-cli.ts#L125-L129

Reply:
But if you don't catch it here, just run the method, and we're not catching anything in `runReactNativeCiCli` then it will be catched in `run` as you expect, or am I missing something?

Reply:
oh, my idea was for `validate()` to throw a very brief error message with the reason (here `only supported in expo projects`) so that it can be displayed as hint in multiselect

here I catch it to add some more context to the message, it would be confusing if the script ended with

```
failed to run react-native-ci-cli because of error:
only supported in expo projects
```

Reply:
Oh, ok, si we can have different message based on setup, that's pretty cool.

Reply:
Maybe just leave some short comment desribing why this error is catched and thrown again.

Reply:
added in cd4eb882c5c705183a0c8370f89c9e15c676e1bb